lib/oauth2_basic_authenticator.rb

```diff
@@ -113,7 +113,8 @@ def json_walk(result, user_json, prop, custom_path: nil)
       path = path.gsub(".[].", ".").gsub(".[", "[")
       segments = parse_segments(path)
       val = walk_path(user_json, segments)
-      result[prop] = val if val.present?
+      # [] should be nil, false should be false
```

Comment:
There's already a test for the []->nil 
`it "can walk json and handle an empty array" do`